STM32/Filtering/Src/array-math.c

```diff
@@ -0,0 +1,106 @@
+/*! 
+**  \brief     Array math functions
+**  \details   Functions to find the max/average of arrays
+**  \author    Luke Walker
+**  \date      09/04/2024
+**/
+
+#include "array-math.h"
+
+/***************************************************************************************************
+** PUBLIC FUNCTION DEFINITIONS
+***************************************************************************************************/
+
+/*!
+** @brief Returns the maximum value in an array of double
+*/
+double max_element(double arr[], int len) {
+    if(len != 0) {
+        double ret_val = arr[0];
+        for(int i = 1; i < len; i++) {
+            if(arr[i] > ret_val) ret_val = arr[i];
+        }
+        return ret_val;
+    }
+
+    return -1.0;
+}
+
+/*!
+** @brief Returns the average value of an array of double
+*/
+double mean_element(double arr[], int len) {
+    if(len != 0) {
+        double ret_val = arr[0];
+        for(int i = 1; i < len; i++) {
+            ret_val += arr[i];
+        }
+        return ret_val / (double)len;
```

Comment:
This is an unnecessary cast. It is the numerator that defines whether it is an integer division or not.